docs/examples-profiling-vtune.md

```diff
@@ -0,0 +1,19 @@
+# Using `VTune` on Linux
+
+[`VTune Profiler`](https://software.intel.com/en-us/vtune-help) is a popular performance profiling tool that targets both 32-bit and 64-bit x86 architectures. The tool collects profiling data and then presents it, allowing for a variety of performance analysis options. Vtune Profiler is available in both commerical and free options. The free download version is backed by a commuity forum support forum is available [`here`](https://software.intel.com/en-us/vtune/choose-download#standalone). This version is appropriate for detailed analysis of your WASM program. Note for jit support, Wasmtime only supports VTune profiling on linux platforms but other platforms are expected to supported in the future.
+
+VTune support in wasmtime is provided through the jit profiling APIs at [`https://github.com/intel/ittapi`](https://github.com/intel/ittapi). These APIs are provided for code generators to report jit activities. These APIs are implemented in a shared library which wasmtime pulls in and links to when vtune support is specified through the `vtune` cargo feature flag. This feature is not enabled by default. When VTune is run, it links to this shared library. Specifically, Wasmtime depends on the ittapi-rs system crate which uses provides the Rut interface to the C based jit profiling APIs. For jit profiling with VTune Profiler, first you want to make sure the `vtune` feature is enabled. After that, enabling runtime support based on how you are using Wasmtime:
+
+* **Rust API** - you'll want to call the [`Config::profiler`] method with
+  `ProfilingStrategy::VTune` to enable profiling of your wasm modules.
+
+* **C API** - you'll want to call the `wasmtime_config_profiler_set` API with a
+  `WASMTIME_PROFILING_STRATEGY_VTUNE` value.
```

Comment:
I think this may not exist in the C API yet? (should just be a few extra lines here and there for this PR)

Reply:
Lol .. Yes sorry. 👍  I was following your lead and didn't go back and add. I see support here should be added in crates/c-api and will do that next push.